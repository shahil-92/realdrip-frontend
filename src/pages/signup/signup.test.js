import React from 'react';
import { Signup } from '.';
import SignupView from './view';

const mockEvent = target => ({
  preventDefault: jest.fn(), target,
});

const sampleUserData = {
  name: 'tester',
  email: 'tester@gmail.com',
  country: 'testCountry',
  state: 'testState',
  address: 'testAddress',
  password: 'Password123',
  confirmPassword: 'Password123',
  agreement: true,
};

describe('<Signup />', () => {
  const props = {
    dispatch: jest.fn(),
    addingUserStarted: false,
    addingUserResolved: false,
    addingUserError: null,
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  console.log('=alterProps=====',alterProps)
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Signup {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to "confirm you email page" if addingUserResolved is true', () => {
      wrapper.setProps(alterProps({ addingUserResolved: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<Signup {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('GET_INPUT', () => {
      it('should update the state with user input', () => {
        const userInput = 'tester';
        wrapperInstance.getInput(mockEvent({ name: 'name', value: userInput }));
        expect(wrapper.state('name')).toEqual(userInput);
      });
    });
    describe('CLEAR_INPUT_ERRORS', () => {
      it('should clear input errors in the state', () => {
        wrapper.setState({ inputErrors: { name: 'Name is a required field' } });
        wrapperInstance.clearInputErrors();
        expect(wrapper.state('inputErrors').name).toBeUndefined();
      });
    });
    describe('REGISTER', () => {
      it('should set inputErrors if validation fails', () => {
        const wrongEmail = 'testergmail.com';
        wrapper.setState({ email: wrongEmail });
        wrapperInstance.register(mockEvent());
        expect(typeof wrapper.state('inputErrors').email).toEqual('string');
      });
      it('should clear inputErrors and dispatch an actions if validation pass', () => {
        wrapper.setState({
          ...sampleUserData, inputErrors: { email: 'Email is required' }
        });
        wrapperInstance.register(mockEvent());
        expect(wrapper.state('inputErrors').email).toBeUndefined();
        expect(props.dispatch).toHaveBeenCalledTimes(1);
      });
    });

    describe('COMOPONENT_DID_UPDATE', () => {
      it('should update the state if addingUserError prop value changes', () => {
        const submissionError = 'This user already exist'
        expect(wrapper.state('submissionError')).toEqual(null);
        wrapper.setProps({ addingUserError: { message: submissionError } });
        expect(wrapper.state('submissionError')).toEqual(submissionError);
      });
    });
  });
});

describe('<SignupView />', () => {
  const props = {
    addingUserError: null,
    addingUserStarted: false,
    getInput: jest.fn(),
    inputErrors: {},
    location: {
      country: null,
      state: null,
    },
    register: jest.fn(),
    submissionError: null,
  };
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<SignupView {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
