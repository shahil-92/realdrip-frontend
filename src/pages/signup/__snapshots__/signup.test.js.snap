// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> Rendering: should redirect to "confirm you email page" if addingUserResolved is true 1`] = `
<Redirect
  to="/email-confirmation"
/>
`;

exports[`<Signup /> Rendering: should render correctly 1`] = `
<View
  addingUserStarted={false}
  getInput={[Function]}
  inputErrors={Object {}}
  location={
    Object {
      "country": null,
      "state": null,
    }
  }
  register={[Function]}
  submissionError={null}
/>
`;

exports[`<SignupView /> Rendering should render correctly 1`] = `
<div
  className="app-ui"
>
  <div
    className="app-layout-canvas"
  >
    <div
      className="app-layout-container"
    >
      <main
        className="app-layout-content"
      >
        <div
          className="container-fluid p-y-md"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-12"
            >
              <div
                className="col-sm-6 "
              >
                <img
                  alt=""
                  className="img img-responsive"
                  src="/assets/img/photos/unsplash.jpg"
                />
              </div>
              <div
                className="col-sm-6"
              >
                <div
                  className="card-block login-top "
                >
                  <form
                    className="form-horizontal m-t-xs"
                  >
                    <p
                      className="login-head"
                    >
                      Welcome to RealDrip
                    </p>
                    <h4
                      className=" form-title"
                    >
                      Management Registration
                    </h4>
                    <p
                      className="form-text"
                    >
                       Already have a management account? 
                      <span
                        className="login-head"
                      >
                        Sign in
                      </span>
                    </p>
                    <div
                      className="col-sm-10 text-center"
                    >
                      <ErrorText
                        text={null}
                      />
                      <Spinner
                        display={false}
                        height={50}
                        width={50}
                      />
                      <p />
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        className="col-sm-5 form-label"
                        htmlFor="name"
                      >
                        Medical Center Name
                      </label>
                      <label
                        className="col-sm-5 form-label"
                        htmlFor="email"
                      >
                        Medical Center Email
                      </label>
                      <div
                        className="col-sm-5"
                      >
                        <input
                          className="form-control"
                          id="name"
                          name="name"
                          onChange={[MockFunction]}
                          placeholder="Name"
                          type="text"
                        />
                        <ErrorText />
                      </div>
                      <div
                        className="col-sm-5"
                      >
                        <input
                          className="form-control"
                          id="email"
                          name="email"
                          onChange={[MockFunction]}
                          placeholder="Email"
                          type="email"
                        />
                        <ErrorText />
                      </div>
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        className="col-sm-10 form-label"
                        htmlFor="location"
                      >
                        Location
                      </label>
                      <div
                        className="col-sm-5"
                      >
                        <CountryDropdown
                          blacklist={Array []}
                          className="form-control"
                          classes=""
                          defaultOptionLabel="Select Country"
                          disabled={false}
                          id="country"
                          labelType="full"
                          name="country"
                          onBlur={[Function]}
                          onChange={[Function]}
                          priorityOptions={Array []}
                          showDefaultOption={true}
                          value=""
                          valueType="full"
                          whitelist={Array []}
                        />
                        <ErrorText />
                      </div>
                      <div
                        className="col-sm-5"
                      >
                        <RegionDropdown
                          blankOptionLabel="Select Region"
                          className="form-control"
                          classes=""
                          country={null}
                          countryValueType="full"
                          customOptions={Array []}
                          defaultOptionLabel="Select Region"
                          disableWhenEmpty={false}
                          disabled={false}
                          id="state"
                          labelType="full"
                          name="state"
                          onBlur={[Function]}
                          onChange={[Function]}
                          showDefaultOption={true}
                          value=""
                          valueType="full"
                        />
                        <ErrorText />
                      </div>
                    </div>
                    <div
                      className="form-group"
                    >
                      <div
                        className="col-sm-10"
                      >
                        <input
                          className="form-control"
                          id="Address"
                          name="address"
                          onChange={[MockFunction]}
                          placeholder="Address"
                          type="text"
                        />
                        <ErrorText />
                      </div>
                    </div>
                    <div
                      className="form-group"
                    >
                      <label
                        className="col-sm-10 form-label"
                        htmlFor="location"
                      >
                        Password
                      </label>
                      <div
                        className="col-sm-5"
                      >
                        <input
                          className="form-control"
                          id="password"
                          name="password"
                          onChange={[MockFunction]}
                          placeholder="password"
                          type="password"
                        />
                        <ErrorText />
                      </div>
                      <div
                        className="col-sm-5"
                      >
                        <input
                          className="form-control"
                          id="confirmPassword"
                          name="confirmPassword"
                          onChange={[MockFunction]}
                          placeholder="confirm password"
                          type="password"
                        />
                        <ErrorText />
                      </div>
                    </div>
                    <div
                      className="radio"
                    >
                      <label
                        htmlFor="agreement"
                      >
                        <input
                          id="agreement"
                          name="agreement"
                          onChange={[MockFunction]}
                          type="radio"
                          value={true}
                        />
                        <span>
                          I agree to the Terms, Privacy Policy and Fees.
                        </span>
                        <br />
                        <ErrorText />
                      </label>
                    </div>
                    <div
                      className="form-group m-b-0"
                    >
                      <div
                        className="col-xs-8"
                      >
                        <button
                          className="btn form-btn"
                          disabled={false}
                          onClick={[MockFunction]}
                          type="submit"
                        >
                          Create account
                        </button>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
